turn_on_led/turn_on_led/turn_on_led.asm

```diff
@@ -0,0 +1,659 @@
+
+
+;free registers: r16, r30, r31
+
+ldi r16, 0b11111111     ; writing bit pattern 1000 0000 to register 16
+out ddra, r16           ; setup data direction for port A - output
+out ddrc, r16			; setup data direction for port C - output
+
+;prepare game registers
+ldi ballCol, 0b11101111 ; ball column - 0 means that column is displayed
+ldi ballRow, 0b01000000 ; ball row - 1 means that ball is in that raw
+ldi vectorX, 0b00000001 ; direction of vector (100 left / 001 right)
+ldi vectorY, 0b00000010 ; direction of vector (10 up / 01 down)
+
+;initialize block pattern
+ldi blockRow1, 0b10101010
+ldi blockRow2, 0b01010101
+ldi blockRow3, 0b10101010
+ldi blockRow4, 0b01010101
+
+;initialize blocks to hit counter. When counter = 0 then the game is over and player has won
+ldi r16, 16
+mov blockCounter, r16
+
+ldi paddleCol, 0b11000111 ; paddle column (0 - led on, 1 - led off) each bit corresponds to one column 
+
+ldi		r20, 0xff		; left button
+out		ddrg, r20	    ; set port G to input
+
+ldi		r20, 0xff		; right button
+out		ddrb, r20	    ; set port B to input
+
+
+game_start:  ; loading start position, the game will start when both buttons are pressed 
+	call start_position
+	in r20, PING		;read left button
+	cpi r20, 0			;if r20 != 0 then button is pressed
+	breq game_start
+	in r20, PINB		;read right button
+	cpi r20, 0			;if r20 != 0 then button is pressed
+	breq game_start		;go back in loop
+
+main:
+	ldi r25, 12 ; number of display loop iteration before game logic
+	
+display:
+	;section responsible for displaying the game status on the screen
+	;each row is displayed separately in 10ms periods. Then the procedure repeats
+	call row_one
+	call delay_10ms
+
+	call row_two
+	call delay_10ms
+
+	call row_three
+	call delay_10ms
+
+	call row_four
+	call delay_10ms
+	
+	call ball
+	call delay_10ms
+
+	call paddle
+	call delay_10ms
+	;displaying all elements on the screen takes approx. 60ms
+
+	
+	;when the screen is displayed 12 times, 
+	;so 12*60ms = 720ms passed the game logic is checked
+	dec r25
+	cpi r25, 0
+	breq game
+	rjmp display
+
+
+game:
+	call button					; read buttons input
+	
+	call check_ball_left_wall	
+	call check_ball_right_wall
+
+	call check_ball_hit_paddle
+
+	call check_ball_hit_block
+
+	call check_ball_top
+
+	;move the ball according to vector 
+	call movementX				; move ball in X axis
+	call movementY				; move ball in Y axis
+
+	call check_win
+	call check_game_over
+	rjmp main
+
+check_win:
+	;if blockCounter = 0 then the player destoryed all the blocks and the player has won
+	ldi r16, 0
+	cp blockCounter, r16
+	breq win
+	ret
+win:
+	call happy_face
+
+button:
+	in r20, PINB ; read from button on PINB to r20
+	cpi r20, 0   ; check if right button was pressed
+	brne move_paddle_right
+	in r20, PING ; read from button on PING to r20
+	cpi r20, 0 ; check if left button was pressed
+	brne move_paddle_left 
+	ret
+
+movementX:
+	cpi vectorX, 0b00000100 ;if vector is equal this value then move ball left
```

Comment:
The bit pattern `0b00000100` can also be written `(1<<2)` (1 shifted to the left 2 times). You use literal bit patterns a lot throughout your code and it makes it a little easier to read them.